### ISSUE TITLE
Source Sendgrid: paid plan upgrade to cover more endpoints

### ISSUE THREAD

**Issue by igrankova** [airbytehq/airbyte#17310]
## Tell us about the problem you're trying to solve
Investigate what paid plan should we subscribe to cover the next endpoints:
Email API:

Marketing Campaigns:

The costs and other details analysis, approval and implementation.

**Comment by igrankova** [comment]
@YowanR, we discussed the needs with Daryna and see that we need to cover the Messages endpoint in the Email API and it is available on the **Pro 100K plan**: 
100,000 emails/month + $0.001 per additional email
$89.95/month, Taxes may apply
All plans above do not cover this endpoint. To get in touch with the sales representative for receiving some custom plan they allow for the users who send more than 100 000 emails per month:
![image](https://user-images.githubusercontent.com/87977540/192755140-6287ca86-b753-4fca-98e6-2f3d55a0458a.png)

**On the Pro 100K plan we are getting:**
![image](https://user-images.githubusercontent.com/87977540/192755467-8b64f3c5-7072-4acc-8365-bb8602cd533d.png)

If this amount is acceptable, I can upgrade our account to Pro using the card from LastPass or any other you provide.

**Comment by YowanR** [comment]
Let's wait on this one. The extensibility team will work on this connector next so we should time it with them so that if we pay, we don't need to pay for more than a month to get the testing done.
@girarda as FYI

**Comment by girarda** [comment]
@YowanR does this mean we'll be able to test the endpoints for development but not we won't be able to cover it as part of our integration tests?

**Comment by YowanR** [comment]
I just checked Metabase and Sendgrid is quite popular (> 5 Cloud workspaces) so I'm okay paying for this to ensure the quality of the connector. I believe that without paying, we won't be able to cover it for our integration test. @igrankova can you confirm?

**Comment by igrankova** [comment]
@YowanR yes, according to this [message](https://github.com/airbytehq/alpha-beta-issues/issues/227#issuecomment-1258063418) we can't reach needed endpoint on our free plan, so the cheapest plan that covers it is Pro 100K plan. So we have to pay $89.95/month for 100,000 emails/month + $0.001 per additional email.

**Comment by YowanR** [comment]
Here's what I propose. Before we go ahead and pay for it, I'll talk to Chris and see if we can try to get some kind of partnership with Sendgrid. Will follow-up in a few days.

**Comment by KayakinKoder** [comment]
Just wanted to add an outside perspective -- we're in the process of evaluating moving many of our data pipelines to Airbyte, precisely because our current provider has poor and outdated test coverage. E.g. for one of Sendgrid's schema, the pipeline was syncing less than 20% of data due to a small bug in our data pipeline provider's connector; this caused us to send marketing emails to some of our users who had ask to be unsubscribed....not good. I'm glad to see Airbyte discussing ensuring full test coverage for every endpoint in connectors.

**Comment by igrankova** [comment]
@YowanR any news about the partnership?

**Comment by YowanR** [comment]
@igrankova Let's move forward and get the Sendgrid account set up for Pro so we can cover all our tests, please! :)

**Comment by igrankova** [comment]
To upgrade we need to finish account setup.
https://app.sendgrid.com/guide
Step 2. **Authenticate your sending domain**
An authenticated domain improves deliverability and reputation, and is recommended for sending email at scale.
https://app.sendgrid.com/settings/sender_auth/domain/create

**Comment by Rhiyo** [comment]
+1 for messages stream. I'll look into it myself and attempt to make a PR if necessary. For now I'll just try and use the custom source builder.

**Comment by Rhiyo** [comment]
Made a PR for this here: https://github.com/airbytehq/airbyte/pull/60243
